### PR TITLE
Enable lazy-loading of post embeds

### PR DESCRIPTION
**Pull request by westonruter**
* Use `visibility:hidden` instead of `clip: rect(1px, 1px, 1px, 1px);` to hide the iframe while it is loading. When the iframe is lazy-loaded, Chromium does not reliably load the iframe when it is clipped but it does load it when it is made transparent. This also fixes an accessibility problem for keyboard navigation, since links in the clipped iframe could still receive focus even though they could not be seen by sighted users, resulting in confusingly hidden tab stops. Setting the iframe to be `visibility:hidden` ensures that the iframe is loaded consistently across browsers and that the contents of the iframe are not interactable until it is made `visible`.
* Update `wp_iframe_tag_add_loading_attr()` to no longer short-circuit when the iframe is a post embed. 
* Use the HTML API to remove the `style` attribute from post embeds in feeds, instead of doing string replacement.

Trac ticket: https://core.trac.wordpress.org/ticket/58773

# Drafted Commit Message

Embeds: Enable lazy-loading of post embeds and fix keyboard a11y for hidden iframes.

Chrome unreliably loads a lazy-loaded iframe when it is hidden using `clip: rect(1px, 1px, 1px, 1px)`. Instead of using `clip`, a lazy-loaded iframe can also be hidden with `visibility:hidden` which results in it loading not only in Chrome but all other browsers. With this change applied, the hard-coded check to prevent lazy-loading post embeds is now removed. An added benefit to using `visibility:hidden` is that the entire iframe in this case is not interactable, meaning that users navigating the document with the keyboard will not unexpectedly encounter tab stops inside of the hidden iframe, as can happen now with `clip` when the JS fails to reveal the loaded iframe. Note also that the `clip` property is deprecated.

Lastly, when such a post embed iframe is rendered in an RSS feed, the `style` attribute is now removed using the HTML Tag Processor as opposed to using string replacement.

Fixes #58773.
Props westonruter, joemcgill, swissspidy, joedolson, adamsilverstein.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

File: src/wp-includes/embed.php [hunk 1]
```diff
@@ -902,6 +902,7 @@ function wp_filter_oembed_iframe_title_attribute( $result, $data, $url ) {
  * Only filters 'rich' and 'video' response types.
  *
  * @since 4.4.0
+ * @todo This needs to be refactored to leverage the HTML API.
```
Comment (joemcgill) [inline src/wp-includes/embed.php]: Is this going to be done as part of this PR? If not, this should be removed before committing.

**Review by joemcgill — APPROVED** [review]
Nice to see a possible a11y improvement here. Thanks for following up, @westonruter. Once comment about a docblock, but pre-approving the changes.

File: src/wp-includes/embed.php (hunk 1 above)
Comment (westonruter) [reply src/wp-includes/embed.php]: There's a similar todo also committed in core:

https://github.com/WordPress/wordpress-develop/blob/3adbaf8ec020d653dd7e6c949a8b51073e726365/src/wp-includes/media.php#L2218

Comment (westonruter) [reply src/wp-includes/embed.php]:
```suggestion
```

**Review by joemcgill — APPROVED** [review]

**Comment by westonruter** [comment]
Committed to trunk in r58143 (b4889e4)

**Pull request closed without merging**